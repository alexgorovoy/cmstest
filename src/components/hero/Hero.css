
.hero {
  background: 50% no-repeat;
  background-size: cover;
  height: 65vh;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 1em;
}

.hero-text {
    color: #ffff;    
    background-color: rgba(0.5, 0.5, 0.5, 0.5);
    width: 100%;
    padding: 1em;
}

