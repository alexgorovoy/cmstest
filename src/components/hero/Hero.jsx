import React, { Component } from 'react';
import { HeroShape } from '../../shapes/HeroShape';
import './Hero.css';

class Hero extends Component {
    
  render() {
    const { heading, subheading, heroImageUrl} = this.props;
    return (        
        <div className="hero" style={{backgroundImage: `url(${heroImageUrl})`}}>
          <div className="hero-text">
            <h1>{heading}</h1>
            <h4>{subheading}</h4>
          </div>
        </div>
    );
  }
}

Hero.propTypes = HeroShape.isRequired;

Hero.defaultProps = {
  heading: '',
  subheading: '',
  heroImageUrl: ''
}

export default Hero;
