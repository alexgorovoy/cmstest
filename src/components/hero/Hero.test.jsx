import React from 'react';
import renderer from 'react-test-renderer';
import Hero from './Hero';

test('Hero / renders as expected', () => {

    const component = renderer.create(
        <Hero heading="Test" subheading="subheading test" heroImageUrl="https://drive.google.com/uc?id=1svw9VdyX4fyRHd1kggq0akDSafmdOS7L" />
      );
      let hero = component.toJSON();
      expect(hero).toMatchSnapshot();    
});