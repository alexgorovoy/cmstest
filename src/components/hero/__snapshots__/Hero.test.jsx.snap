// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero / renders as expected 1`] = `
<div
  className="hero"
  style={
    Object {
      "backgroundImage": "url(https://drive.google.com/uc?id=1svw9VdyX4fyRHd1kggq0akDSafmdOS7L)",
    }
  }
>
  <div
    className="hero-text"
  >
    <h1>
      Test
    </h1>
    <h4>
      subheading test
    </h4>
  </div>
</div>
`;
